jupyter_server/serverapp.py

```diff
@@ -141,6 +120,51 @@
     urlencode_unix_socket_path,
 )
 
+try:
+    import resource
+except ImportError:
+    # Windows
+    resource = None  # type:ignore[assignment]
+
+from jinja2 import Environment, FileSystemLoader
+from jupyter_core.paths import secure_write
+
+from jupyter_server.transutils import _i18n, trans
+from jupyter_server.utils import ensure_async, pathname2url, urljoin
+
+# the minimum viable tornado version: needs to be kept in sync with setup.py
+MIN_TORNADO = (6, 1, 0)
+
+try:
+    import tornado
+
+    assert tornado.version_info >= MIN_TORNADO
+except (ImportError, AttributeError, AssertionError) as e:  # pragma: no cover
+    raise ImportError(_i18n("The Jupyter Server requires tornado >=%s.%s.%s") % MIN_TORNADO) from e
+
+
+try:
```

Comment:
This block is repeated from above

Reply:
I have no idea how that happened... Fixed 😆